Author: aszlig <aszlig@nix.build>
Date:   Fri Mar 30 20:29:49 2018 +0200

    Remove community guidelines
    
    The standalone version is invite-only anyways and the community
    guidelines really vary from installation to installation, so let's get
    rid of that.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: remove-community-guidelines.patch

diff --git a/website/client/components/chat/chatCard.vue b/website/client/components/chat/chatCard.vue
index b4daaf2ef7..8da4868b57 100644
--- a/website/client/components/chat/chatCard.vue
+++ b/website/client/components/chat/chatCard.vue
@@ -19,7 +19,7 @@ div
       .action.d-flex.align-items-center(v-if='!inbox', @click='copyAsTodo(msg)')
         .svg-icon(v-html="icons.copy")
         div {{$t('copyAsTodo')}}
-      .action.d-flex.align-items-center(v-if='(inbox || (user.flags.communityGuidelinesAccepted && msg.uuid !== "system")) && (!isMessageReported || user.contributor.admin)', @click='report(msg)')
+      .action.d-flex.align-items-center(v-if='(inbox || (msg.uuid !== "system")) && (!isMessageReported || user.contributor.admin)', @click='report(msg)')
         .svg-icon(v-html="icons.report", v-once)
         div(v-once) {{$t('report')}}
       .action.d-flex.align-items-center(v-if='msg.uuid === user._id || inbox || user.contributor.admin', @click='remove()')
diff --git a/website/client/components/groups/chat.vue b/website/client/components/groups/chat.vue
index 424fc71e33..f6e76095ad 100644
--- a/website/client/components/groups/chat.vue
+++ b/website/client/components/groups/chat.vue
@@ -1,10 +1,10 @@
 <template lang="pug">
   .row.chat-row
     .col-12
-      h3.float-left.label(:class="{accepted: communityGuidelinesAccepted }") {{ label }}
+      h3.float-left.label {{ label }}
       div.float-right(v-markdown='$t("markdownFormattingHelp")')
 
-      .row(v-if="communityGuidelinesAccepted")
+      .row
         textarea(:placeholder='placeholder',
                   v-model='newMessage',
                   ref='user-entry',
@@ -29,14 +29,12 @@
                 :caretPosition = 'caretPosition',
                 :chat='group.chat')
 
-      community-guidelines
-
       .row.chat-actions
         .col-6.chat-receive-actions
           button.btn.btn-secondary.float-left.fetch(v-once, @click='fetchRecentMessages()') {{ $t('fetchRecentMessages') }}
           button.btn.btn-secondary.float-left(v-once, @click='reverseChat()') {{ $t('reverseChat') }}
         .col-6.chat-send-actions
-          button.btn.btn-primary.send-chat.float-right(:disabled="!communityGuidelinesAccepted", @click='sendMessage()') {{ $t('send') }}
+          button.btn.btn-primary.send-chat.float-right(@click='sendMessage()') {{ $t('send') }}
 
       slot(
         name="additionRow",
@@ -51,7 +49,6 @@
   import debounce from 'lodash/debounce';
 
   import autocomplete from '../chat/autoComplete';
-  import communityGuidelines from './communityGuidelines';
   import chatMessage from '../chat/chatMessages';
   import { mapState } from 'client/libs/store';
   import markdownDirective from 'client/directives/markdown';
@@ -63,7 +60,6 @@
     },
     components: {
       autocomplete,
-      communityGuidelines,
       chatMessage,
     },
     data () {
@@ -87,9 +83,6 @@
       currentLength () {
         return this.newMessage.length;
       },
-      communityGuidelinesAccepted () {
-        return this.user.flags.communityGuidelinesAccepted;
-      },
     },
     methods: {
       // https://medium.com/@_jh3y/how-to-where-s-the-caret-getting-the-xy-position-of-the-caret-a24ba372990a
@@ -261,10 +254,6 @@
   .chat-row {
     position: relative;
 
-    .label:not(.accepted) {
-      color: #a5a1ac;
-    }
-
     .row {
       margin-left: 0;
       margin-right: 0;
diff --git a/website/client/components/groups/group.vue b/website/client/components/groups/group.vue
index 2695b6fac5..d4f43348e0 100644
--- a/website/client/components/groups/group.vue
+++ b/website/client/components/groups/group.vue
@@ -375,9 +375,6 @@ export default {
     },
   },
   methods: {
-    acceptCommunityGuidelines () {
-      this.$store.dispatch('user:set', {'flags.communityGuidelinesAccepted': true});
-    },
     async load () {
       if (this.isParty) {
         this.searchId = 'party';
diff --git a/website/client/components/groups/tavern.vue b/website/client/components/groups/tavern.vue
index 8915ef24d8..bbdfb4f9ac 100644
--- a/website/client/components/groups/tavern.vue
+++ b/website/client/components/groups/tavern.vue
@@ -10,7 +10,6 @@
     chat(
       :label="$t('tavernChat')",
       :group="group",
-      :placeholder="$t('tavernCommunityGuidelinesPlaceholder')",
       @fetchRecentMessages="fetchRecentMessages()"
     )
   .col-12.col-sm-4.sidebar
@@ -101,8 +100,6 @@
 
       sidebar-section(:title="$t('helpfulLinks')")
         ul
-          li
-           router-link(to='/static/community-guidelines', v-once) {{ $t('communityGuidelinesLink') }}
           li
             router-link(to="/groups/guild/f2db2a7f-13c5-454d-b3ee-ea1f5089e601") {{ $t('lookingForGroup') }}
           li
diff --git a/website/client/components/userMenu/profile.vue b/website/client/components/userMenu/profile.vue
index dba76d8a65..81ebebcb8d 100644
--- a/website/client/components/userMenu/profile.vue
+++ b/website/client/components/userMenu/profile.vue
@@ -83,8 +83,6 @@
     .row(v-if='editing')
       h1 {{$t('editProfile')}}
       .col-12
-        .alert.alert-info.alert-sm(v-html='$t("communityGuidelinesWarning", managerEmail)')
-
         .form-group
           label {{ $t('displayName') }}
           input.form-control(type='text', :placeholder="$t('fullName')", v-model='editingProfile.name')
diff --git a/website/client/router/index.js b/website/client/router/index.js
index cd97a355fd..246d9214f0 100644
--- a/website/client/router/index.js
+++ b/website/client/router/index.js
@@ -10,7 +10,6 @@ const HomePage = () => import(/* webpackChunkName: "entry" */'client/components/
 
 const AppPage = () => import(/* webpackChunkName: "static" */'client/components/static/app');
 const ClearBrowserDataPage = () => import(/* webpackChunkName: "static" */'client/components/static/clearBrowserData');
-const CommunityGuidelinesPage = () => import(/* webpackChunkName: "static" */'client/components/static/communityGuidelines');
 const FAQPage = () => import(/* webpackChunkName: "static" */'client/components/static/faq');
 const FeaturesPage = () => import(/* webpackChunkName: "static" */'client/components/static/features');
 const GroupPlansPage = () => import(/* webpackChunkName: "static" */'client/components/static/groupPlans');
@@ -251,7 +250,6 @@ const router = new VueRouter({
       children: [
         { name: 'app', path: 'app', component: AppPage, meta: {requiresLogin: false}},
         { name: 'clearBrowserData', path: 'clear-browser-data', component: ClearBrowserDataPage, meta: {requiresLogin: false}},
-        { name: 'communityGuidelines', path: 'community-guidelines', component: CommunityGuidelinesPage, meta: {requiresLogin: false}},
         { name: 'faq', path: 'faq', component: FAQPage, meta: {requiresLogin: false}},
         { name: 'features', path: 'features', component: FeaturesPage, meta: {requiresLogin: false}},
         { name: 'groupPlans', path: 'group-plans', component: GroupPlansPage, meta: {requiresLogin: false}},
diff --git a/website/server/libs/user/index.js b/website/server/libs/user/index.js
index 5737a9ed6c..c4321265cc 100644
--- a/website/server/libs/user/index.js
+++ b/website/server/libs/user/index.js
@@ -39,7 +39,6 @@ const updatablePaths = [
   'flags.showTour',
   'flags.tour',
   'flags.tutorial',
-  'flags.communityGuidelinesAccepted',
   'flags.welcomed',
   'flags.cardReceived',
   'flags.warnedLowHealth',
diff --git a/website/server/models/user/schema.js b/website/server/models/user/schema.js
index 75efb7f937..4e8c3db98e 100644
--- a/website/server/models/user/schema.js
+++ b/website/server/models/user/schema.js
@@ -227,7 +227,6 @@ let schema = new Schema({
     // Used to enable weekly recap emails as users login
     lastWeeklyRecapDiscriminator: Boolean,
     onboardingEmailsPhase: String, // Keep track of the latest onboarding email sent
-    communityGuidelinesAccepted: {$type: Boolean, default: false},
     cronCount: {$type: Number, default: 0},
     welcomed: {$type: Boolean, default: false},
     armoireEnabled: {$type: Boolean, default: true},
